### ISSUE TITLE
add-anchor - greedily matching too much?

### ISSUE THREAD

**Issue by rm-hull** [weavejester/codox#145]
I have a docstring which contains the following:

```
  --8<-- unnecessary stuff elided --8<-- 

  ...source of the quotation may be given using the cite attribute, while a text
  representation of the source can be given using the <cite> element. Example:

    <blockquote cite=\"http://www.worldwildlife.org/who/index.html\">
      For 50 years, WWF has been protecting the future of nature. The
      world's leading conservation organization, WWF works in 100
      countries and is supported by 1.2 million members in the United
      States and close to 5 million globally.
    </blockquote>")
```

Using the latest codox version (0.10.1, but also happened on 0.10.0), it renders as:

<img width="558" alt="screen shot 2016-10-10 at 01 51 07" src="https://cloud.githubusercontent.com/assets/1915543/19235253/795cde32-8ee8-11e6-93db-20db1c2f2a2a.png">

I would've expected the [add-anchor and its url-regex](https://github.com/weavejester/codox/blob/c27255b70513da1775f7f5794eabceaa2e98b618/codox/src/codox/writer/html.clj#L51-L56) to not include the ending double-quote or angle bracket, and was initially puzzled where the semi-colon comes from...

Looking into it, I think the actual cause is that the content is HTML-escaped _before_ the `add-anchor` method is run, so that the actual thing that is passed through to the regex matcher is something like:

```
&lt;blockquote cite=&quot;http://www.worldwildlife.org/who/index.html&quot;&gt;
```

Not really sure what the best solution would be here though, as prior HTML-escaping is the right thing to do, but then fixing the regex to not match the `&` and beyond will be tricky as `&` is obviously valid in a URL query.

**Comment by weavejester** [comment]
Yeah, I'm not certain either. Perhaps `"` on the end of the URL is ignored if the URL comes directly after a preceding `"`.
